src/test/java/org/opensearch/ad/transport/SearchTopAnomalyResultResponseTests.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.ad.transport;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+
+import org.opensearch.ad.TestHelpers;
+import org.opensearch.common.io.stream.BytesStreamOutput;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.test.OpenSearchTestCase;
+
+public class SearchTopAnomalyResultResponseTests extends OpenSearchTestCase {
+
+    public void testSerialization() throws IOException {
+        SearchTopAnomalyResultResponse originalResponse = new SearchTopAnomalyResultResponse(
+            Arrays.asList(TestHelpers.randomAnomalyResultBucket())
+        );
+
+        BytesStreamOutput output = new BytesStreamOutput();
+        originalResponse.writeTo(output);
+        StreamInput input = output.bytes().streamInput();
+        SearchTopAnomalyResultResponse parsedResponse = new SearchTopAnomalyResultResponse(input);
+        assertEquals(originalResponse.getAnomalyResultBuckets(), parsedResponse.getAnomalyResultBuckets());
+    }
+
+    public void testEmptyResults() {
```

Comment:
nit: I assume this is fine as is since there would be an exception or compiler error if you couldn't initialize an empty response but maybe add an assertEquals with expected empty response. Same for `testPopulatedResults` below.

Reply:
Prefer to not add an assertEquals here since it's a bit redundant. The response constructor that takes in a list (used here and in `testPopulatedResults`) simply assigns it to the single `anomalyResultBuckets` field, so not much to check.